test/test_kmeans.py

```diff
@@ -15,26 +15,19 @@ def test_fails_for_repeating_centroids():
     data = pd.read_csv('/Users/amruk/PycharmProjects/kmeans/setA.txt', delimiter=' ')
     file.close()
     data1 = data[["Gsto1", "Gstm1", "Cd9", "Prdx1", "Coro1a", "Rac2", "Perp"]].values
-    for i in range(10):
+    i = 2
+    while i <= 10:

```

Comment:
Don't make it a loop here. If it fails, how do you know what were the conditions of failure? Was it for 5 or 8 clusters? Rather use [parameterized tests](https://docs.pytest.org/en/latest/parametrize.html), so the test framework keeps track of everything for you.